django/core/checks/urls.py

```diff
@@ -64,16 +64,26 @@ def check_pattern_startswith_slash(pattern):
     """
     Check that the pattern does not begin with a forward slash.
     """
-    regex_pattern = pattern.regex.pattern
-    if regex_pattern.startswith('/') or regex_pattern.startswith('^/'):
-        warning = Warning(
-            "Your URL pattern {} has a regex beginning with a '/'. "
-            "Remove this slash as it is unnecessary.".format(describe_pattern(pattern)),
-            id="urls.W002",
-        )
-        return [warning]
-    else:
-        return []
+    errors = []
+
+    if not hasattr(pattern, 'regex'):
+        errors.append(Warning("url objects should have a property regex",
```

Comment:
This message should include some helpful information to debug the error. Currently it gives no information as to where the problem is. This is the idea behind the existing `describe_pattern()` function.